proto/net.proto

```diff
@@ -210,6 +275,9 @@ message NetworkFormat {
     NETWORK_SE_WITH_HEADFORMAT = 4;
     NETWORK_ONNX = 5;
     NETWORK_ATTENTIONBODY_WITH_HEADFORMAT = 6;
+    NETWORK_CLASSICAL_WITH_MULTIHEADFORMAT = 131;
+    NETWORK_SE_WITH_MULTIHEADFORMAT = 132;
+    NETWORK_ATTENTIONBODY_WITH_MULTIHEADFORMAT = 134;
```

Comment:
I think a `NETWORK_MULTIHEADFORMAT_FLAG = 128` is also needed?

Reply:
I find this solution somewhat ugly
```diff
    NETWORK_SE_WITH_HEADFORMAT = 4;
    NETWORK_ONNX = 5;
    NETWORK_ATTENTIONBODY_WITH_HEADFORMAT = 6;
+   NETWORK_CLASSICAL_WITH_MULTIHEADFORMAT = 131;
+   NETWORK_SE_WITH_MULTIHEADFORMAT = 132;
+   NETWORK_ATTENTIONBODY_WITH_MULTIHEADFORMAT = 134;
```
but don't immediately know anything nicer..

Maybe just this
```diff
    NETWORK_SE_WITH_HEADFORMAT = 4;
    NETWORK_ONNX = 5;
    NETWORK_ATTENTIONBODY_WITH_HEADFORMAT = 6;
+   NETWORK_CLASSICAL_WITH_MULTIHEADFORMAT = 7;
+   NETWORK_SE_WITH_MULTIHEADFORMAT = 8;
+   NETWORK_ATTENTIONBODY_WITH_MULTIHEADFORMAT = 9;
```

and a few helper functions in C++ code:
```cpp
Format GetBaseCanonicalWhateverFormat(Format x) {
   switch(x) {
    case NETWORK_CLASSICAL_WITH_MULTIHEADFORMAT: return NETWORK_CLASSICAL;
    case NETWORK_SE_WITH_MULTIHEADFORMAT: return NETWORK_SE;
   default:
    return x
   }
}

bool HasMultihead(Format x) {
   switch(x) {
  case NETWORK_CLASSICAL_WITH_MULTIHEADFORMAT:
  vase NETWORK_SE_WITH_MULTIHEADFORMAT:
  case NETWORK_ATTENTIONBODY_WITH_MULTIHEADFORMAT:
      return true;
  default:
       return false;
  }
}
```

Reply:
We can drop the flag (current code doesn't need it a lot), but I don't think we can change the numbers, too many nets out in the wild with this. 
We can maybe also drop `NETWORK_CLASSICAL_WITH_MULTIHEADFORMAT` and `NETWORK_SE_WITH_MULTIHEADFORMAT`, it looked like a good idea at the time (guilty) to convert all formats to the corresponding multihead format when loading, but now I'm not so sure (see https://github.com/LeelaChessZero/lc0/pull/1915 for our current approach).

Reply:
Ok, maybe keep numbers then, but in the C++ code, don't do the "`& 128`" thing but rather have explicit functions?

Reply:
Agreed, the `& 128` should go. This is why I suggested the flag, but there are other ways.

Reply:
Done.